HW_TEST

michallow opened the pull request.

michallow committed:
HW_TEST

KamilMr commented on routes/api/contacts.js:
```diff
@@ -19,79 +20,89 @@ const favoriteSchema = Joi.object({
-    const contacts = await listContacts();
-    res.status(200).json(contacts);
+    const contact = await getContactById(req.params.id, req.user._id);
+    if (!contact || contact.owner.toString() !== req.user._id) {
```
Ten drugi warunek, po `||` jest nie potrzebny. Jeżeli funkcja w linii 25 zwróci kontakt to znaczy że znalazła tę osobę z takim id i takim kluczem owner. Ten pierwszy ma sens. 

Natomiast cała ta metoda jest niepotrzebna, gdyż metoda getContactById już sprawdza czy ten user ma kontakt.

KamilMr commented on routes/api/contacts.js:
```diff
@@ -19,79 +20,89 @@ const favoriteSchema = Joi.object({
 });
 
-router.delete("/:id", async (req, res, next) => {
+router.get('/:id', auth, checkOwner, async (req, res, next) => {
```
```suggestion
router.get('/:id', auth, async (req, res, next) => {
```

KamilMr replied:
Jak pisalem przy omawianiu tej funkcji, ona jest niepotrzebna.

KamilMr commented on routes/api/users.js:
```diff
@@ -0,0 +1,88 @@
+
+router.get('/logout', auth, async (req, res, next) => {
+  try {
+    const user = await User.findById(req.user.id);
```
Tego bloku już nie potrzebujesz poniewaz `auth` juz sprawdziło czy dany user ma prawo dostepu do tego miejsca. I jesli tu dotarl to istnieje :)

KamilMr commented on routes/api/users.js:
```diff
@@ -0,0 +1,88 @@
+    const user = await User.findById(req.user.id);
+    if (!user) {
+      return res.status(401).json({ message: 'Not authorized' });
+    }
```
Wykorzystaj req do tego aby pobrać dane, nie pytaj ponownie bazy danych gdyż w `auth` już to zrobiłeś

KamilMr reviewed (approved):
Popracuj na przyszłość nad nazywaniem commitów do pull request. W tym zadaniu widzę że jest jeden i nazwa nic nie sugeruje.

michallow committed:
HW_4_Corrections

routes/api/contacts.js
- removal of unnecessary condition
routes/api/users.js
- simplifying the code by removing unnecessary block
